<launch>    
    <param name="reset_state" type="int" value="1" />
    <param name="test_phase" type="int" value="3" />
    
    <node name="testing" pkg="race" type="testing" output="screen"  required="true"/>
        <param name="SCARA_MODE" type="bool" value="true" />

        <param name="calibration_x_intercept" value="0.7669" />
        <param name="calibration_y_intercept" value="-0.0927" />
        <param name="calibration_z_intercept" value="0" />
        <param name="calibration_x" value="6.0207" />
        <param name="calibration_y" value="6.408" />
        <param name="calibration_z" value="6.3" />

        <param name="max_xy" value="5.2" />
        <param name="min_xy" value="0.06" />
        <param name="max_z" value="0.13" />
        <param name="acc_xy" value="0.02" />
        <param name="acc_zz" value="0.002" />
        <param name="maxUP_xy" value="7" />
        <param name="accUP_xy" value="0.04" />

        <param name="fod_xy" value="0.2" />   <!--fraction of deceleration: start decel-->
        <param name="fod_z" value="0.5" />   <!--fraction of deceleration: start decel-->
        <param name="kp_xy" value="0.8" /> <!--p gain for x- y-direction control-->
        <param name="kp_z" value="1" />  <!--p gain for z-direction control-->

        <param name="x_tol_margin" value="0.8" /> <!-- x tolerance critical value -->
        <param name="y_tol_margin" value="0.8" /> <!-- y tolerance critical value -->
        <param name="z_tol_margin" value="0.005" /> <!-- z tolerance critical value -->
    <!-- pixel label: x = 97 ~ 395 / y = 38 ~ 473 -->
        <param name="pixel_Xmin" value="150" />
        <param name="pixel_Xmax" value="390" /> 
        <param name="pixel_Ymax" value="460" />
        <param name="pixel_Ymin" value="35" />

        <param name="x_tf_cali" value="0.8" />
        <param name="y_tf_cali" value="0.8" />
        <param name="x_tf_intercept" value="-8.1" />
        <param name="y_tf_intercept" value="-10.5" />

    <node name="reset" pkg="race" type="reset" output="screen" respawn="true" respawn_delay="3"/>

    <node name="serial_server" pkg="rosserial_server" type="serial_node" output="screen" respawn="true" respawn_delay="3">
        <param name="port" value="/dev/ttyUSB0" />
        <param name="baud" value="115200" />
        
        <rosparam>
            require:
                publishers: [ mecanum_fromSTM, reset_fromSTM, scaraflag_fromSTM ]
                subscribers: [ mecanum_toSTM, intake_toSTM, scara_toSTM ]
        </rosparam> 
    </node>

</launch>